monai/transforms/io/array.py

```
@@ -209,10 +209,12 @@ def __init__(
                 try:
                     self.register(the_reader(*args, **kwargs))
-                except OptionalImportError:
-                    warnings.warn(
-                        f"required package for reader {_r} is not installed, or the version doesn't match requirement."
-                    )
+                except OptionalImportError as e:
+                    raise RuntimeError(
+                        f"The required package for reader '{_r}' is not installed, or the version doesn't match "
+                        f"the requirement. If you want to use '{_r}', please install the required package. "
+                        f"If you want to use an alternative reader, do not specify the `reader` argument."
+                    ) from e
```

Ruff (0.15.4)

[warning] Avoid specifying long messages outside the exception class

(TRY003)